### PR TITLE
fix(vaults): return correct collateral after liquidation

### LINKED ISSUE #7779: Liquidation collateral proceeds overpaying to highest CR vault following auction completion
## Describe the bug

When testing [happy path liquidation](https://docs.google.com/document/d/1bXnQ_o3wwfnu-uhX2KSb6ZXB0dU6CsAsGBWm1DXgmE0/edit?usp=sharing) after several addressing PRs #7766 and #7767 were merged - I discovered that the liquidation collateral proceeds were still distributed to the vault with the highest CR once auction completes instead of being split proportionally across each vault.

## To Reproduce
1. With ATOM price set to 12.34 Set up 3x vaults and 3x bids per happy path liquidation scenario (link)
2. Change price to 9.99 (pushPrice 9.99) before price lock period to lock in price for liquidation triggering
3. Observe auction to completion (check progress throughout with inter auction status)
4. Check Vaults UI where you created the vaults - they should be liquidated.
5. Observe highest CR vault has receieved 9.34 ATOM whereas the other two liquidated vaults receive 0.

## Expected behavior
Highest CR vault should not receive all the liquidated collateral proceeds, should be divided proportionally amongst the 3 vaults.

## Platform Environment
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
fixes: #7779
refs: #7346
refs: #7767

## Description

The debt to be subtracted from the vault's value is the current debt level divided by the realized price.

### Security Considerations

None

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

Added a unit test to follow the scenario carefully.

### CODE REVIEW

File: packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js [hunk 1]
```diff
@@ -58,7 +58,9 @@ export const defaultParamValues = debtBrand =>
  * @param {Array<NatValue> | Ratio} priceOrList
  * @param {RelativeTime} quoteInterval
  * @param {Amount | undefined} unitAmountIn
- * @param {bigint} [startFrequency]
+ * @param {object} actionParams
+ * @param {bigint} actionParams.startFrequency
+ * @param {bigint} actionParams.discountStep
```
Comment (turadg) [inline packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js]: +1 to using the params object. but please use the `AuctionParams` type.
```suggestion
 * @param {Pick<AuctionParams, 'StartFrequency' | 'DiscountStep'>} actionParams
```
Note that test-vaultFactory depends on this function signature too

File: packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js [hunk 2]
```diff
@@ -108,7 +110,7 @@ export const setupElectorateReserveAndAuction = async (
     ClockStep: 2n,
     StartingRate: 10500n,
     LowestRate: 5500n,
-    DiscountStep: 2000n,
+    DiscountStep: discountStep || 2000n,
```
Comment (turadg) [inline packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js]: the function signature said this is required. I think that's best but if you want to have a default param value please express express it in the signature.

**Review by turadg — COMMENTED** [review]
Code style review. I'll wait for passing tests to do full

File: packages/inter-protocol/src/vaultFactory/vaultManager.js [hunk 3]
```diff
@@ -702,8 +708,10 @@ export const prepareVaultManagerKit = (
                 debtAmount,
                 vault.getCurrentDebt(),
               );
-              const vaultDebt = floorMultiplyBy(debtAmount, debtPortion);
-              const collatPostDebt = AmountMath.subtract(vCollat, vaultDebt);
+              const vaultDebt = ceilDivideBy(debtAmount, price);
```
Comment (turadg) [inline packages/inter-protocol/src/vaultFactory/vaultManager.js]: elsewhere in this function "debt" is in the stable token, so I found this a little confusing.
```suggestion
              const collatOwed = ceilDivideBy(debtAmount, price);
```
but not worth another CI build if this one passes

File: packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js [hunk 4]
```diff
@@ -2183,6 +2206,215 @@ test('Bug 7422 vault reinstated with no assets', async t => {
   });
 });
 
+test('Bug 7346 excess collateral to holder', async t => {
```
Comment (turadg) [inline packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js:2209]: confirmed that this test fails without the vaultManager changes
```
  Bug 7346 excess collateral to holder

  Difference (- actual, + expected):

    {
      brand: Object @Alleged: aEth brand {},
  -   value: 8239n,
  +   value: 3165n,
    }
```

**Review by turadg — APPROVED** [review]
🚀

File: packages/inter-protocol/src/vaultFactory/vaultManager.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/vaultFactory/vaultManager.js]: done.  changed to `debtInCollateral`.

File: packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js]: done

File: packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/test/vaultFactory/vaultFactoryUtils.js]: thanks for catching that. done.

**Pull request merged**